src/Catel.Test/Catel.Test.Shared/Collections/FastObservableCollectionFacts.cs

```diff
@@ -144,5 +145,141 @@ public void ReturnsSingleElementUsingLinq()
                 Assert.AreEqual(42, allInts);
             }
         }
+
+        [TestFixture]
+        public class TheSuspensionModeMethod
+        {
+            [TestCase]
+            public void AddingItemsInAddingMode()
+            {
+                var counter = 0;
+                var eventArgs = (NotifyCollectionChangedEventArgs)null;
+
+                var fastCollection = new FastObservableCollection<int>();
+                fastCollection.AutomaticallyDispatchChangeNotifications = false;
+                fastCollection.CollectionChanged += (sender, e) =>
+                    {
+                        counter++;
+                        eventArgs = e;
+                    };
+
+                using (fastCollection.SuspendChangeNotifications(SuspensionMode.Adding))
+                {
+                    fastCollection.Add(1);
+                    fastCollection.Add(2);
+                    fastCollection.Add(3);
+                    fastCollection.Add(4);
+                    fastCollection.Add(5);
+                }
+
+                Assert.AreEqual(1, counter);
+                Assert.AreEqual(NotifyCollectionChangedAction.Add, eventArgs.Action);
+                CollectionAssert.AreEqual(eventArgs.NewItems, new[] { 1, 2, 3, 4, 5 });
+            }
+
+            [TestCase]
+            public void MultipleActionsWithoutSuspendingNotifications()
```

Comment:
If Remove is called and the item was before the items, just decrease the indices. If Add is called and the item was before already added items (an insert), just increase the indices.

Then, you check if the Added already contains the item being removed, if so, remove it again. If the Removed contains the added one, it's now a modification.

I think it's really that simple, but might be overlooking something. But again, we can look into this in phase 2, happy to postpone.